mungegithub/mungers/submit-queue.go

```diff
@@ -1149,6 +1149,26 @@ func (sq *SubmitQueue) doGithubE2EAndMerge(obj *github.MungeObject) bool {
 		glog.Infof("Skipping retest since head and base sha match previous attempt!")
 		atomic.AddInt32(&sq.retestsAvoided, 1)
 	} else {
+		if sq.retestPR(obj) {
+			return true
+		}
+	}
+
+	if !sq.e2eStable(true) {
+		if sq.validForMerge(obj) {
+			sq.interruptedObj = newInterruptedObject(obj)
+		}
+		sq.SetMergeStatus(obj, e2eFailure)
+		return true
+	}
+
+	sq.mergePullRequest(obj)
+	return true
+}
+
+// Returns true if we can discard the PR from the queue, false if we must keep it for later.
+func (sq *SubmitQueue) retestPR(obj *github.MungeObject) bool {
+	if len(sq.RequiredRetestContexts) > 0 {
```

Comment:
style nit: Invert this logic to `return false` if `len(sq.RequiredRetestContexts) == 0` and remove a level on indentation. Then I'll happy.